src/System.Linq/src/System/Linq/OrderedEnumerable.cs

```diff
@@ -46,9 +54,16 @@ public TElement[] ToArray()
 
             TElement[] array = new TElement[count];
             int[] map = SortedMap(buffer);
-            for (int i = 0; i != array.Length; i++)
+            try
+            {
+                for (int i = 0; i != array.Length; i++)
+                {
+                    array[i] = buffer._items[map[i]];
+                }
+            }
+            finally
```

Comment:
There should be no scope for an exception here bar something very catastrophic (thread abort). The try…finally is perhaps not needed.